### ISSUE TITLE
P3: Run eval harness on pilot runs and verify flip-rate

### ISSUE THREAD

**Issue by riteshsivaraman** [riteshsivaraman/shortcut-learning-lstm#19]
## Task

Once Person 2 has completed the two pilot training runs (baseline and 50%-end, seed 42), run all four evaluation modes against those checkpoints and sanity-check that the metrics are sensible. In particular, verify that the flip-rate for the 50%-end model is meaningfully higher than for the baseline — if it isn't, the trigger injection may be broken or the model too small to learn the shortcut.

## Acceptance criteria

- [ ] `all_metrics()` runs cleanly against `results/lstm_baseline_seed42/model.pt`
- [ ] `all_metrics()` runs cleanly against `results/lstm_strength_50_end_seed42/model.pt`
- [ ] Baseline flip-rate is near 0 (trigger never seen in training, so model should not have learned it)
- [ ] 50%-end flip-rate is noticeably above baseline (confirms shortcut was learned)
- [ ] Both runs' metrics appended to `results/results.csv` with all four mode columns populated

## Blockers

- [ ] `no_trigger`, `trigger_injected`, and `flip_rate` eval modes implemented — #18
- [ ] Person 2 has completed both pilot runs (baseline + 50%-end, seed 42) — #17

## How to run

```bash
python scripts/train.py --config configs/lstm_baseline.yaml --seed 42
python scripts/train.py --config configs/lstm_strength_50_end.yaml --seed 42
python -c "
import torch
from src.data.dataset import load_imdb
from src.models.lstm import LSTMClassifier
from src.eval.metrics import all_metrics

# load model and test set, then:
# metrics = all_metrics(model, test_dataset, trigger_id, trigger_position='end')
# print(metrics)
"
```
